netket/hilbert/continuous_space_particles.py

```diff
@@ -0,0 +1,69 @@
+from typing import Tuple
+
+from .abstract_hilbert import AbstractHilbert
+
```

Comment:
I think it is worth turning this into a dataclass?

Reply:
Hilbert spaces are not dataclasses but they are dataclass-compatible if you properly define `_attr`.
`_attr` should return a tuple of all fields that must be used to compute the hash and equality among hillbert spaces of the same type.


You can make it a dataclass but 1) it would be useless (for now), because hilbert is always passed around as a `static_argnum` object, and 2) you should be extra careful not to break comparison between hilbert spaces and hashability.